module_5/src/com/epam/dragontreasure/entity/treasure/Treasure.java

```diff
@@ -0,0 +1,47 @@
+package com.epam.dragontreasure.entity.treasure;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+
+public abstract class Treasure implements Comparable<Treasure> {
+
+    protected Enum type;
+    protected int numberOfItems;
+
+    public Treasure(Enum type, int numberOfItems) throws NullPointerException, IllegalArgumentException {
```

Comment:
maybe let's make this constructor protected to deny creating abstract Treasure objects?
```suggestion
    protected Treasure(Enum type, int numberOfItems) throws NullPointerException, IllegalArgumentException {
```